### ISSUE TITLE
Find a correct way to load all js and css assets

### ISSUE THREAD

**Issue by philippejadin** [agorakit/agorakit#473]
I'd like to experiment with basset https://github.com/Laravel-Backpack/basset to load all the assets, in order to remove dependency on CDN's

**Comment by linc** [comment]
It would be nice to have a way to specify a CDN location if the host requires one, but I agree overall it's best the default not be a third-party CDN. If I had to choose between "always loads from third-party" and "must load from local install" I'd still choose the latter.

**Comment by philippejadin** [comment]
I'd like to find a way to simply download the needed file locally on build or I don't know when precisely in the process of distributing the app (we could even put the library in the repository but it sounds a bit ugly). At least find a way that is both easy to use as developers and provide a ready to use package. Don't want to reinvent npm also :-)

**Comment by linc** [comment]
After thinking about this more, I'd probably just put the files in the repo. When I think about the complexity of these other systems (nom, Phing, Basset), we're basically talking about adding a dependency to manage a dependency that doesn't really need much managing. It's OK to just commit it and replace it when we need to, and that's a simple thing anyone can do. Updating dependency trees & updating Phing routines is kinda hard by comparison.

We could add a minification optimization later when it's helpful.

**Comment by philippejadin** [comment]
Agreed : it's simple to do and I did like that in the beginning. Using random cdns was a temporary measure to quickly test things anyway. This is a good first issue, tagging it.

Also note that only bootstrap (tabler actually) css and unpoly js is loaded upfront, the other libraries are loaded on demand when a matching element is inserted in the dom using unpoly compilers.
There is a [loadJs()](https://github.com/agorakit/agorakit/blob/main/public/js/compilers.js#L27) and a [loadCSS()](https://github.com/agorakit/agorakit/blob/main/public/js/compilers.js#L52) function for that purpose

**Comment by philippejadin** [comment]
Note for later, this looks nice : https://symfony.com/doc/current/frontend/asset_mapper.html#importmaps-javascript

**Comment by zerline** [comment]
I agree, `importmap` looks nice.  

Looking at this technology, leads me to considerations about SASS. Why do you use Sass and not just CSS? 
cf https://gomakethings.com/is-it-time-to-drop-sass/

In both cases (importmap and SASS vs CSS being rather new features), I also wonder how the display works on older browsers.

**Comment by philippejadin** [comment]
I'm also now considering that no build at all is the best way. And it is currently what we are doing in fact :-)

(I mentionned assetmapper from symfony as an alternative to nodejs based tools, but the trend going build less is more tempting.)

How it currently works : 

- tabler.io (based on bootstrap), fontawesome and unpoly  are loaded in the [header blade template](https://github.com/agorakit/agorakit/blob/main/resources/views/partials/header.blade.php#L12).
- when another library is needed (for example a calendar must be shown), this is all handled using [unpoly compilers](https://unpoly.com/up.compiler)
    - a js function to load external js on demand is here : https://github.com/agorakit/agorakit/blob/main/public/js/compilers.js#L27
    - same for css : https://github.com/agorakit/agorakit/blob/main/public/js/compilers.js#L52

- here is for example the function that loads js calendar when there is a calendar in a view : https://github.com/agorakit/agorakit/blob/main/public/js/compilers.js#L266

Note that in blade views, if you want to trigger the calendar unpoly compiler, you just add .js-calendar class to an element to compile this specific element to a calendar.

**Comment by zerline** [comment]
Thank you very much for this explanation.

I'm currently working on accessibility, and some of the loaded javascript code would need edits for this concern.

For instance, the `body` form field cannot be used with keyboard, or with difficulty, and it cannot at all be used with a screenreader. The `<textarea>` tag is totally disabled (display: None).

**Comment by philippejadin** [comment]
I suppose we can translate that to "summernote is not accessible" ? 

Some hints here : https://github.com/summernote/summernote/issues?q=is%3Aissue+accessibility+is%3Aclosed

Note that summernote is very unfortunately minimally maintained, a shame because 
- it's very lightweight
- it's very easy to handle fileuploads and user mentions with @ 

If there is indeed a specific accessibility issue with summernote, don't hesitate to create a separate issue for that.

**Comment by zerline** [comment]
There's a specific accessibility issue. Possibly not with summernote though (the `<textarea..` code is not from summernote as far as I understand).
